#! /usr/bin/env python
#Autogenerated V4L Dynamic Control

PACKAGE='mrpt_localization'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("debug", bool_t, 0, "Debug", False)
enum_motion_noise_type = gen.enum([gen.const("gaussian", int_t, 0, "gaussian"), gen.const("Thrun", int_t, 1, "Thrun")], "motion type")
gen.add("motion_noise_type", int_t, 0, "motion noise", 0, 0, 1, edit_method=enum_motion_noise_type)
gen.add("gaussian_alpha_1", double_t, 0, "alpha_1", 0.034, 0, 1)
gen.add("gaussian_alpha_2", double_t, 0, "alpha_2", 0.057, 0, 1)
gen.add("gaussian_alpha_3", double_t, 0, "alpha_3", 0.014, 0, 1)
gen.add("gaussian_alpha_4", double_t, 0, "alpha_4", 0.097, 0, 1)
gen.add("gaussian_alpha_xy", double_t, 0, "alpha_xy", 0.005, 0, 1)
gen.add("gaussian_alpha_phi", double_t, 0, "alpha_phi", 0.05, 0, 3.14)

gen.add("use_default_motion", bool_t, 0, "use default motion in case of missing odom data", False)
gen.add("default_noise_xy", double_t, 0, "default_noise_xy", 0.1, 0, 1)
gen.add("default_noise_phi", double_t, 0, "default_noise_phi", 2.0, 0, 3.14)

gen.add("update_while_stopped", bool_t, 0, "keep updating filter while the robot is stopped", True)

exit(gen.generate(PACKAGE, "mrpt_localization", "Motion"))
